salt/state.py

```diff
@@ -2002,12 +2005,17 @@ def reconcile_procs(self, running):
                                'changes': {}}
                     try:
                         with salt.utils.fopen(ret_cache, 'rb') as fp_:
-                            ret = msgpack.loads(fp_.read())
+                            ret = msgpack.loads(fp_.read(), encoding='utf-8')
```

Comment:
If this is a file being read from the file system, yes, we're better off with the `__salt_system_encoding__`. Of course, its not full proof but it also does not require the user to be forced to use utf-8 for his system.